### PR TITLE
📖  update and supplement webhook documentation with example to work with core types

### PR DESCRIPTION
**Pull request by camilamacedo86**
Closes: https://github.com/kubernetes-sigs/kubebuilder/pull/3399
also, closes: https://github.com/kubernetes-sigs/kubebuilder/issues/1241

### CODE REVIEW

File: docs/book/src/reference/webhook-for-core-types.md
Comment (mythi) [inline docs/book/src/reference/webhook-for-core-types.md:134]: IIRC I had to add something like https://github.com/mythi/kubebuilder/blob/2db6ee6f869b69ba46501b734e50c41a438bc5d6/docs/book/src/reference/webhook-for-core-types.md?plain=1#L47-L50 to get it working.

File: docs/book/src/reference/webhook-for-core-types.md
Comment (camilamacedo86) [inline docs/book/src/reference/webhook-for-core-types.md:47]:
```suggestion
// +kubebuilder:webhook:path=/mutate--v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io
```

File: docs/book/src/reference/webhook-for-core-types.md
Comment (camilamacedo86) [reply docs/book/src/reference/webhook-for-core-types.md:134]: The same marker is in the line 47 
What exactly is missing could you please let me know?

Comment (mythi) [reply docs/book/src/reference/webhook-for-core-types.md:134]: I had this clarification added: "(for webhooks the `path` is of format `/mutate-<group>-<version>-<kind>`. Since this documentation uses `Pod` from the core API group, the group needs to be an empty string)"

the marker is: `/mutate--v1-pod`

Comment (camilamacedo86) [reply docs/book/src/reference/webhook-for-core-types.md]: done good catcher

**Review by mythi — APPROVED** [review]
this looks good and is very informative, thanks

**Comment by camilamacedo86** [comment]
/hold

We need to get merged: https://github.com/kubernetes-sigs/kubebuilder/pull/4060
So that we can have the convey and then have the examples here updated accordingly.

**Review by camilamacedo86 — COMMENTED** [review]
We are for a good time trying to fix this doc :-) 
It has a LGTM
And the examples with CustomInterfaces are respecting the comments and the changes addressed in #4060.

So, it shows good enough to fly 🚀  since it is better than before. Therefore, moving forward with this one.

However, further improvements are welcome always.
If you see that need changes please, feel free to raise an PR with your suggestions.

**Comment by mythi** [comment]
> So, it shows good enough to fly 🚀 since it is better than before. Therefore, moving forward with this one.

+1, thanks!

**Pull request merged**
